src/addr.rs

```diff
@@ -148,31 +148,31 @@ impl VirtAddr {
 
     /// Returns the 12-bit page offset of this virtual address.
     #[inline]
-    pub fn page_offset(&self) -> PageOffset {
+    pub fn page_offset(self) -> PageOffset {
```

Comment:
I think this could be a breaking change, but I'm not sure. If it is breaking, I don't think that the change is worth it because the potential performance improvement is very small, especially with inlining.